src/components/Commentable/index.tsx

```diff
@@ -1,29 +1,45 @@
 import * as React from 'react';
 import makeClassName from 'classnames';
 
+import AddComment, { PublicProps as AddCommentProps } from './AddComment';
 import { AnyReactNode } from '../../typeUtils';
 import styles from './styles.module.scss';
 
 export type PublicProps = {
+  addCommentClassName?: string;
   as: React.ReactType;
-  children?: AnyReactNode;
+  children: (addComment: JSX.Element) => AnyReactNode;
   className?: string;
   id?: string;
-};
+  shellRef?: (element: HTMLElement | null) => void;
+} & AddCommentProps;
 
 // This represents something that can be commented on, such as a line of code.
 const CommentableBase = ({
+  addCommentClassName,
   as: AsComponent,
   children,
   className,
+  fileName,
   id,
+  line,
+  shellRef,
+  versionId,
 }: PublicProps) => {
   return (
     <AsComponent
       className={makeClassName(styles.commentable, className)}
       id={id}
+      ref={shellRef}
     >
-      {children}
+      {children(
+        <AddComment
```

Comment:
This is nice the way you got `Commentable` to include `AddComment`. Well done! 👍

Reply:
Well, the downside is you can't do normal shallow rendering when testing a component that uses `Commentable`. This is pretty major and it's why I originally didn't want to do it. However, we know how to write test helpers to reach in and evoke callbacks so, yeah, I guess it's OK. The more I thought about it, it felt like the *right* thing to do.